builder/azure/dtl/builder.go

```diff
@@ -95,7 +102,7 @@ func (b *Builder) Run(ctx context.Context, ui packersdk.Ui, hook packersdk.Hook)
 		return nil, err
 	}
 	if b.config.ClientConfig.ObjectID == "" {
-		b.config.ClientConfig.ObjectID = getObjectIdFromToken(ui, spnCloud)
+		b.config.ClientConfig.ObjectID = *objectId
```

Comment:
I think it did but I forgot to respond my bad!

I think we don't because of how the upstream method works, NewAzureClient returns a pointer to a string that we instantiate, if that's null it should just resolve to an empty string, then below this we check if object ID is set to empty, and alert the user that "could not determine the ObjectID for the user, which is required for WIndows builds", we don't fail the builder if we can't determine the object ID, although we probably should for Windows builds as it is required